### PR TITLE
feat: Update backend to Unarchive Assessment

### PR DESCRIPTION
**Pull request by carissa-tang**
## Notion ticket link

[Update backend to Unarchive Assessment](https://www.notion.so/uwblueprintexecs/Update-backend-to-Unarchive-Assessment-6c8ccb9494dc4bb0b1be65f6d218962d?pvs=4)

## Implementation description
* Implemented function to unarchive an `archived` test
   1. Duplicate given test and set to draft
   2. Soft delete archived test
   3. Rollback changes if delete fails
* Wrote unit tests (success + fail)

## Steps to test
1. yarn test

## What should reviewers focus on?
* I talked to Brian abt moving the soft / hard delete logic into the service directly (rather than in the resolver) so I can use that functionality in my function

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/models/user.model.ts [hunk 1]
```diff
@@ -64,6 +64,7 @@ const UserSchema: Schema = new Schema({
   },
 });
 
+/* eslint-disable func-names */
```
Comment (carissa-tang) [inline backend/models/user.model.ts:67]: https://medium.com/@lucasdavidferrero/dont-use-arrow-functions-when-you-use-mongoose-schema-method-190b79f1640c

File: backend/services/implementations/__tests__/testService.test.ts [hunk 2]
```diff
@@ -133,4 +133,33 @@ describe("mongo testService", (): void => {
...
+      {
+        ...mockTest,
+        status: AssessmentStatus.DRAFT,
+      },
+      unarchivedTest,
+    );
+    expect(test.id).not.toEqual(unarchivedTest.id);
+
+    const originalTest = await MgTest.findById(test.id);
+    // TODO: update this to be soft delete instead of hard delete
```
Comment (carissa-tang) [inline backend/services/implementations/__tests__/testService.test.ts:157]: follow up ticket for once brian merges his PR https://www.notion.so/uwblueprintexecs/Update-unarchiveTest-unit-test-40608ab551c64370a6837e3d7d122961?pvs=4

File: backend/services/implementations/testService.ts [hunk 3]
```diff
@@ -146,6 +146,51 @@ class TestService implements ITestService {
...
+      }
+      if (test.status !== AssessmentStatus.ARCHIVED) {
+        throw new Error(`Test ID ${id} is not in archived status`);
+      }
+
+      unarchivedTest = test;
+      // eslint-disable-next-line no-underscore-dangle
+      unarchivedTest._id = mongoose.Types.ObjectId();
+      unarchivedTest.isNew = true;
+      unarchivedTest.status = AssessmentStatus.DRAFT;
```
Comment (carissa-tang) [inline backend/services/implementations/testService.ts]: if duplicates also become draft, i can directly call `duplicateTest` here

File: backend/services/implementations/testService.ts [hunk 4]
```diff
@@ -147,6 +147,46 @@ class TestService implements ITestService {
...
+
+    try {
+      const test = await MgTest.findById(id);
+      if (!test) {
+        throw new Error(`Test ID ${id} not found`);
+      }
+      if (test.status !== AssessmentStatus.ARCHIVED) {
+        throw new Error(`Test ID ${id} is not in archived status`);
+      }
+
```
Comment (carissa-tang) [inline backend/services/implementations/testService.ts]:
```suggestion
```

**Review by joyce-shi — APPROVED** [review]
LGTM

**Pull request merged**